src/dfx/src/commands/wallet/mod.rs

```diff
@@ -48,6 +48,9 @@ enum SubCommand {
     Custodians(custodians::CustodiansOpts),
     Deauthorize(deauthorize::DeauthorizeOpts),
     Name(name::NameOpts),
+    #[deprecated(
+        note = "This command is deprecated and will be removed in a future version. Eventually, the wallet canister will become an dfx extension."
```

Comment:
This should say what command they should use instead.  The deprecation warning doesn't need to mention the intention to move the functionality to an extension.

Also, we should display the deprecation warning at runtime.  As far as I can tell, this is only displayed at compile time.

```suggestion
        note = "This command is deprecated and will be removed in a future version. Use XXXXXX instead."
```


```bash
$ dfx-wip wallet redeem-faucet-coupon abc
WARN: Trying to redeem a wallet coupon on a local replica. Did you forget to use '--network ic'?
Redeeming coupon. This may take up to 30 seconds...
```

Reply:
right... 🤦